hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/BlockManagerImpl.java

```diff
@@ -222,9 +224,11 @@ public AllocatedBlock allocateBlock(final long size, ReplicationType type,
       }
 
       if (null == pipeline) {
-        // TODO: #CLUTIL Make the selection policy driven.
-        pipeline = availablePipelines
-            .get((int) (Math.random() * availablePipelines.size()));
+        Map<String, Object> params = new HashMap<>();
+        params.put(
+            PipelineChoosePolicy.PIPELINE_CHOOSE_POLICY_PARAM_SIZE, size);
+        pipeline = pipelineChoosePolicy.choosePipeline(
```

Comment:
It seems to be a standard parameter which is added to all the implementation. Wouldn't be better to add it as a type-safe field? Or we can use a `PipelineRequestInformation` class instead of the `Map<String,Object>` if you would like to make it easier to add more information latest.